spring-data-jpa/src/test/java/org/springframework/data/jpa/repository/query/HqlQueryTransformerTests.java

```diff
@@ -1043,6 +1047,40 @@ void createsCountQueryUsingAliasCorrectly() {
 		assertCountQuery("select distinct a, count(b) as c from Employee GROUP BY n","select count(distinct a, count(b)) from Employee AS __ GROUP BY n");
 	}
 
+	@Test // GH-3427
+	void sortShouldBeAppendedWithSpacingInCaseOfSetOperator() {
+
+        String source = "SELECT tb FROM Test tb WHERE (tb.type='A') UNION SELECT tb FROM Test tb WHERE (tb.type='B')";
+		String target = createQueryFor(source, Sort.by("Type").ascending());
```

Comment:
After this line `target` has the value 
```sql
SELECT tb 
    FROM Test tb 
    WHERE (tb.type = 'A') 
    order by tb.Type asc 
UNION 
SELECT tb 
    FROM Test tb 
    WHERE (tb.type = 'B') 
    order by tb.Type asc`
```
i.e. the `order by` is applied twice which is illegal, since according to answers here https://stackoverflow.com/questions/4715820/how-to-order-by-with-union-in-sql the later `order by` applies to the full `union` and an `order by` on the individual selects might be even illegal (probably depends on the database)

Reply:
since we do not know the target db nor how it handles order by we cannot imply ordering the last is the correct behaviour, nor do we have the means to define which of the selects to decorate. an alternative would be to raise an error and demand wrapping the  entire thing as it's done in one of the follow up tests.

Reply:
Probably, let's wait and see what kind of reports we get. We can then still react to the feedback we receive.

Reply:
reading through IBM an Oracle SQL reference docs @schauder has a good point in the above mentioned query not being compliant (at least not in the form it is currently rendered)

> For example, the following is not valid (SQLSTATE 428FJ):
>`SELECT * FROM T1  `
>`ORDER BY C1       `
>`UNION             `
>`SELECT * FROM T2  `
>`ORDER BY C1       `
>
>
> The following example is valid:
> `(SELECT * FROM T1 `
> `ORDER BY C1)      `
> `UNION             `
> `(SELECT * FROM T2 `
> `ORDER BY C1)      `